core/shared/src/main/scala/scalaz/zio/Queue.scala

```diff
@@ -355,34 +354,35 @@ object ZQueue {
     final def offer(a: A): UIO[Boolean] = offerAll(List(a))
 
     final def offerAll(as: Iterable[A]): UIO[Boolean] =
-      for {
-        _ <- checkShutdownState
-
-        remaining <- IO.effectTotalWith { platform =>
-                      val pTakers                = if (queue.isEmpty()) unsafePollN(takers, as.size) else List.empty
-                      val (forTakers, remaining) = as.splitAt(pTakers.size)
-                      (pTakers zip forTakers).foreach {
-                        case (taker, item) => unsafeCompletePromise(taker, item, platform)
+      ZIO.suspend {
+        for {
+          _ <- checkShutdownState
+
+          remaining <- IO.effectTotal {
+                        val pTakers                = if (queue.isEmpty()) unsafePollN(takers, as.size) else List.empty
+                        val (forTakers, remaining) = as.splitAt(pTakers.size)
+                        (pTakers zip forTakers).foreach {
+                          case (taker, item) => unsafeCompletePromise(taker, item)
+                        }
+                        remaining
                       }
-                      remaining
-                    }
-
-        added <- if (remaining.nonEmpty) {
-                  // not enough takers, offer to the queue
-                  for {
-                    surplus <- IO.effectTotalWith { platform =>
-                                val as = unsafeOfferAll(queue, remaining.toList)
-                                unsafeCompleteTakers(platform)
-                                as
-                              }
-                    res <- if (surplus.isEmpty) IO.succeed(true)
-                          else
-                            strategy.handleSurplus(surplus, queue) <* IO.effectTotalWith(
-                              platform => unsafeCompleteTakers(platform)
-                            )
-                  } yield res
-                } else IO.succeed(true)
-      } yield added
+
+          added <- if (remaining.nonEmpty) {
+                    // not enough takers, offer to the queue
+                    for {
+                      surplus <- IO.effectTotal {
+                                  val as = unsafeOfferAll(queue, remaining.toList)
+                                  unsafeCompleteTakers
```

Comment:
You need to add `()` otherwise dotty complains.
Nice simplification!

Reply:
Thank you!